lib/AST/Decl.cpp

```diff
@@ -4604,7 +4605,8 @@ ValueDecl *ProtocolDecl::getSingleRequirement(DeclName name) const {
 }
 
 AssociatedTypeDecl *ProtocolDecl::getAssociatedType(Identifier name) const {
-  auto results = const_cast<ProtocolDecl *>(this)->lookupDirect(name);
+  const auto flags = NominalTypeDecl::LookupDirectFlags::IgnoreNewExtensions;
+  auto results = const_cast<ProtocolDecl *>(this)->lookupDirect(name, flags);
```

Comment:
Same comment as above; why pass in `IgnoreNewExtensions`? Would a comment help? The temp variable `flags` is  bit odd. I'm guessing you used it because 4609 would be longer without it, but IMO it doesn't help comprehension.